pkg/controller/replication/stringslock.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package replication
+
+import (
+	"sync"
+)
+
+type stringMutex struct {
+	lock  sync.Mutex
```

Comment:
Can you explain the use case you need this particular lock if you already have a look in `StringsLock` that exclusively grants access to `strings`?

Reply:
As discussed in person, we can replace the `stringMutex` struct and use `lock.TryLock()` instead.